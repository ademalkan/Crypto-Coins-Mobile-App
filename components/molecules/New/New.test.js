import React from "react";
import renderer from "react-test-renderer";
import New from "./New";

jest.mock("@react-navigation/native", () => ({
  useNavigation: () => ({
    navigate: jest.fn(),
  }),
}));

describe("New", () => {
  it("should render the news item correctly", () => {
    const item = {
      title: "News Title",
      description: "News Description",
      author: "John Doe",
      thumb_2x: "news.png",
      url: "https://example.com/news",
    };

    const component = renderer.create(<New item={item} />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
