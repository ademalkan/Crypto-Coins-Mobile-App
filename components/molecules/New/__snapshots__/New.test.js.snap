// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`New should render the news item correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "white",
        "borderRadius": 8,
        "marginBottom": 16,
        "padding": 16,
      }
    }
  >
    <Image
      resizeMode="contain"
      source={1}
      style={
        {
          "height": 200,
          "padding": 6,
          "width": 300,
        }
      }
    />
    <Text
      style={
        {
          "fontSize": 18,
          "fontWeight": "bold",
          "marginBottom": 8,
        }
      }
    >
      News Title
    </Text>
    <Text
      style={
        {
          "fontSize": 16,
          "marginBottom": 8,
        }
      }
    >
      News Description
    </Text>
    <Text
      style={
        {
          "color": "gray",
          "fontSize": 14,
        }
      }
    >
      John Doe
    </Text>
  </View>
</View>
`;
